lib/netuitive_ruby_api/data_manager.rb

```diff
@@ -0,0 +1,142 @@
+      else
+        server_interaction { netuitived_server.addSample(metric_id, val) }
+      end
+    end
+
+    def add_counter_sample(metric_id, val)
+      if @sample_cache_enabled
+        NetuitiveRubyApi::NetuitiveLogger.log.debug "adding sample to cache: #{metric_id}"
+        sample_cache_addition { data_cache.add_counter_sample(metric_id: metric_id, val: val) }
+      else
+        server_interaction { netuitived_server.addCounterSample(metric_id, val) }
+      end
+    end
+
+    def aggregate_metric(metric_id, val)
+      if @sample_cache_enabled
+        NetuitiveRubyApi::NetuitiveLogger.log.debug "adding sample to cache: #{metric_id}"
+        sample_cache_addition { data_cache.add_aggregate_metric(metric_id: metric_id, val: val) }
+      else
+        server_interaction { netuitived_server.aggregateMetric(metric_id, val) }
+      end
+    end
+
+    def aggregate_counter_metric(metric_id, val)
+      if @sample_cache_enabled
+        NetuitiveRubyApi::NetuitiveLogger.log.debug "adding sample to cache: #{metric_id}"
+        sample_cache_addition { data_cache.add_aggregate_counter_metric(metric_id: metric_id, val: val) }
+      else
+        server_interaction { netuitived_server.aggregateCounterMetric(metric_id, val) }
+      end
+    end
+
+    def event(message, timestamp, title, level, source, type, tags)
+      if @event_cache_enabled
+        NetuitiveRubyApi::NetuitiveLogger.log.debug "adding event to cache: #{message}"
+        event_cache_addition do
+          data_cache.add_event(message: message,
+                               timestamp: timestamp,
+                               title: title,
+                               level: level,
+                               source: source,
+                               type: type,
+                               tags: tags)
+        end
+      else
+        server_interaction { netuitived_server.event(message, timestamp, title, level, source, type, tags) }
+      end
+    end
+
+    def exception_event(exception, klass, tags)
+      NetuitiveRubyApi::ErrorLogger.guard('error during exception_event') do
+        hash = { message: exception.message }
+        hash[:backtrace] = exception.backtrace.join("\n\t") if (defined? exception.backtrace) && !exception.backtrace.nil?
+        if @event_cache_enabled
+          NetuitiveRubyApi::NetuitiveLogger.log.debug "adding exception event to cache: #{hash[:message]}"
+          event_cache_addition do
+            data_cache.add_exception_event(exception: hash,
+                                           klass: klass,
+                                           tags: tags)
+          end
+        else
+          server_interaction { netuitived_server.exceptionEvent(hash, klass, tags) }
+        end
+      end
+    end
+
+    def flush_samples
+      NetuitiveRubyApi::ErrorLogger.guard('error during flush_samples') do
+        sample = data_cache.clear_sample_cache
```

Comment:
I'll add to this. Travis will catch anything up to the point where we make a drb call to the daemon, but if that call fails we'll still lose samples. As it stands that's the current functionality of the api, but we can make another issue to discuss some sort of drb error handling. I'm kind of 50/50 on it, because I want to keep the max cache size setting pretty strict.